### DISCUSSION TITLE
Creating Dags using Custom API

### DISCUSSION THREAD

**Discussion by mashirali** [apache/airflow#21178] [Q&A]
I want to make an api to create dags dynamically (Single or multiple). I have created/written the API code for it and put this on endpoints.py file located at ```airflow/www/api/experimental/endpoints.py```. But when i run airflow, postman says address not found even i have given the same route address. 

For this i have compiled the project using ./breeze on Pycharm.

Any help and guidance regarding this will be appreciated.

**Comment by potiuk** [comment]
My comment - there is no point in writing API for that. Creating DAGs is equivalent to putting file in the folder - Scheduler will have to anyhow scan the folder in order for the DAGs to be actually visible in UI and scheduled. There is no other way currently in Airlfow to do it. There is no point in adding this functionality to Airflow (the API runs in webserver not in scheduler) and using experimental API for that is even worse idea. 

If you REALLY want to create DAGs via API, write your own webserver that simply creates the file - it can use any technology, flask or not and it should just create files in DAGs folder.  DON'T integrate it with Airflow. It's pretty useless and you ask yourself for troubles for upgrades (experimental API is going to be removed sooner or later for example).
